PhoneBook.Dejmenek/Controllers/ContactsController.cs

```diff
@@ -0,0 +1,207 @@
+
+        if (ContactNotExist(contactToUpdate))
+        {
+            AnsiConsole.MarkupLine("There are currently no contacts available. Please create some contacts before updating a contact.");
+            return;
+        }
+
+        if (_userInteractionService.GetConfirmation("Do you want to update the name? (y/n)"))
+        {
+            contactToUpdate.Name = _userInteractionService.GetContactName();
+        }
+
+        if (_userInteractionService.GetConfirmation("Do you want to update the category? (y/n)"))
+        {
+            List<Category> categories = _categoriesRepository.GetCategories();
+            if (categories.Count == 0)
+            {
+                AnsiConsole.MarkupLine("There are currently no categories available.");
+            }
+            else
+            {
+                string categoryName = _userInteractionService.GetCategory(Mapper.ToCategoryDTOs(categories));
+                Category chosenCategory = categories.Single(c => c.Name == categoryName);
+
+                contactToUpdate.CategoryId = chosenCategory.Id;
+            }
+        }
+
+        if (_userInteractionService.GetConfirmation("Do you want to update the phone number? (y/n)"))
+        {
+            string phoneNumber = _userInteractionService.GetPhoneNumber();
+
+            while (_contactsRepository.PhoneNumberExists(phoneNumber))
+            {
+                phoneNumber = _userInteractionService.GetPhoneNumber();
+            }
+
+            contactToUpdate.PhoneNumber = phoneNumber;
+        }
+
+        if (_userInteractionService.GetConfirmation("Do you want to update the email? (y/n)"))
+        {
+            contactToUpdate.Email = _userInteractionService.GetEmail();
+        }
+
+        _contactsRepository.UpdateContact(contactToUpdate);
+    }
+
+    public void DeleteContact()
+    {
+        Contact contactToDelete = GetContact();
+
+        if (ContactNotExist(contactToDelete))
+        {
+            AnsiConsole.MarkupLine("There are currently no contacts available. Please create some contacts before deleting a contact.");
+            return;
+        }
+
+        _contactsRepository.DeleteContact(contactToDelete.Id);
+    }
+
+    public List<ContactDTO> GetContactsByCategory()
+    {
+        List<Category> categories = _categoriesRepository.GetCategories();
+
+        if (categories.Count == 0)
+        {
+            return [];
+        }
+
+        string categoryName = _userInteractionService.GetCategory(Mapper.ToCategoryDTOs(categories));
+        int categoryId = categories.Single(cat => cat.Name == categoryName).Id;
+
+        var contacts = _contactsRepository.GetContactsByCategory(categoryId);
+
+        if (contacts.Count == 0)
+        {
+            return [];
+        }
+
+        return Mapper.ToContactDTOs(contacts);
+    }
+
+    public Contact GetContact()
+    {
+        List<Contact> contacts = _contactsRepository.GetAllContacts();
+
+        if (contacts.Count == 0)
+        {
```

Comment:
This would be heavily frowned upon in a professional environment. It's very misleading. You're actually returning a contact when no contacts exist. It's better to return null and let the caller deal with it. 🤔